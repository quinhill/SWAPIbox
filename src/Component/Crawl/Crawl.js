import React from 'react';
import './crawl.css';
import PropTypes from 'prop-types';

const Crawl = ({title, crawlText, date}) => {
  return (
    <section class="star-wars">

      <div class="crawl">

        <div class="title">
          <h1>{title}</h1>
        </div>
        <p>{crawlText}</p>
        <p>{date}</p>
      </div>

    </section>
  )
}

Crawl.Proptypes = {
  title: PropTypes.string,
  crawlText: PropTypes.string,
  date: PropTypes.string
}

export default Crawl;