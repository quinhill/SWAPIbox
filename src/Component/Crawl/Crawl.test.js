import React from 'react';
import Crawl from './Crawl';
import { shallow } from 'enzyme';

describe('Crawl', () => {

  let wrapper;
  let mockTitle;
  let mockCrawlText;
  let mockDate;

  beforeEach(() => {
    mockTitle = 'string'
    mockCrawlText = 'string'
    mockDate = 'string'
    wrapper = shallow(<Crawl 
                        title={mockTitle}
                        crawlText={mockCrawlText}
                        date={mockDate}
                        />)
  })
  
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})