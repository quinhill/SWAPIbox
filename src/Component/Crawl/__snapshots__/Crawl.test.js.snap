// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Crawl
    crawlText="string"
    date="string"
    title="string"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        class="crawl"
>
        <div
                class="title"
        >
                <h1>
                        string
                </h1>
        </div>
        <p>
                string
        </p>
        <p>
                string
        </p>
</div>,
      "class": "star-wars",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            class="title"
>
            <h1>
                        string
            </h1>
</div>,
          <p>
            string
</p>,
          <p>
            string
</p>,
        ],
        "class": "crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              string
</h1>,
            "class": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "string",
            },
            "ref": null,
            "rendered": "string",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "string",
          },
          "ref": null,
          "rendered": "string",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "string",
          },
          "ref": null,
          "rendered": "string",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          class="crawl"
>
          <div
                    class="title"
          >
                    <h1>
                              string
                    </h1>
          </div>
          <p>
                    string
          </p>
          <p>
                    string
          </p>
</div>,
        "class": "star-wars",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              class="title"
>
              <h1>
                            string
              </h1>
</div>,
            <p>
              string
</p>,
            <p>
              string
</p>,
          ],
          "class": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                string
</h1>,
              "class": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "string",
              },
              "ref": null,
              "rendered": "string",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "string",
            },
            "ref": null,
            "rendered": "string",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "string",
            },
            "ref": null,
            "rendered": "string",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
